[Bug] NIFTI datatypeCode 64 is not yet supported

### Describe the Bug

When I use `cornerstone-js` to load NIFTI2 files using the `nifti-volume-loader`, I get the error `NIFTI datatypeCode 64 is not yet supported` for certain NIFTI files.




### Steps to Reproduce

1. Using the `packages/nifti-volume-loader/example/niftiBasic` example, I modify the path to a locally served NIFTI file with float 64 type.
2. I serve the file locally: `http-server -p 8080 --cors`, `npm install -g http-server` if required.
3. I run the examples: `yarn build-and-serve-static-examples`, and navigate to `niftiBasic`
4. The console logs return:
```bash
createNiftiImageIdsAndCacheMetadata.ts:216 Error reading Nifti header: Error: NIFTI datatypeCode 64 is not yet supported
    at getArrayConstructor (dataTypeCodeHelper.ts:21:13)
    at handleNiftiHeader (createNiftiImageIdsAndCacheMetadata.ts:201:49)
    at appendData (createNiftiImageIdsAndCacheMetadata.ts:128:21)
    at processChunk (createNiftiImageIdsAndCacheMetadata.ts:111:5)
    at createNiftiImageIdsAndCacheMetadata.ts:88:9
    at step (cornerstoneNiftiImageLoader.ts:283:1)
    at Object.next (cornerstoneNiftiImageLoader.ts:283:1)
    at asyncGeneratorStep (cornerstoneNiftiImageLoader.ts:283:1)
    at _next (cornerstoneNiftiImageLoader.ts:283:1)
```


### The current behavior

It is not possible to load NIFTI2 volumes. Trivially, I converted a NIFTI1 to NIFTI2 volume using the following python script:

```python
#!/usr/bin/env python3
"""
Convert NIFTI file to float64 datatype for bug reproduction

This script takes an existing NIFTI file and converts it to float64 (datatype 64)
to reproduce cornerstone3D NIFTI loader issues with 64-bit data types.

Usage:
    python convert_to_float64.py input.nii.gz output_float64.nii.gz
"""

import sys
import argparse
import nibabel as nib
import numpy as np


def convert_nifti_to_float64(input_path, output_path):
    """
    Convert a NIFTI file to float64 datatype
    
    Args:
        input_path: Path to input NIFTI file
        output_path: Path for output NIFTI file with float64 data
    """
    print(f"Loading NIFTI file: {input_path}")
    
    # Load the NIFTI file
    nii = nib.load(input_path)
    
    # Get the image data
    data = nii.get_fdata()
    
    print(f"Original shape: {data.shape}")
    print(f"Original dtype: {data.dtype}")
    print(f"Original data range: {data.min():.6f} to {data.max():.6f}")
    
    # Convert to float64
    data_float64 = data.astype(np.float64)
    
    print(f"New dtype: {data_float64.dtype}")
    print(f"New data range: {data_float64.min():.6f} to {data_float64.max():.6f}")
    
    # Create new NIFTI image with float64 data
    # Use the original header as a template but update datatype
    new_header = nii.header.copy()
    
    # Set the datatype to float64 (code 64 in NIFTI spec)
    new_header.set_data_dtype(np.float64)
    
    print(f"NIFTI datatype code: {new_header['datatype']}")
    print(f"Bits per voxel: {new_header['bitpix']}")
    
    # Create new NIFTI image
    new_nii = nib.Nifti1Image(data_float64, nii.affine, new_header)
    
    # Save the new NIFTI file
    print(f"Saving float64 NIFTI file: {output_path}")
    nib.save(new_nii, output_path)
    
    # Verify the saved file
    print("\nVerifying saved file...")
    verify_nii = nib.load(output_path)
    verify_header = verify_nii.header
    
    print(f"Verified datatype code: {verify_header['datatype']}")
    print(f"Verified bits per voxel: {verify_header['bitpix']}")
    print(f"Verified data dtype: {verify_nii.get_fdata().dtype}")
    
    print(f"\nSuccessfully converted {input_path} to float64 format!")
    print(f"Output saved as: {output_path}")
    print(f"This file should trigger the cornerstone3D 'datatype 64 not supported' error")


def main():
    parser = argparse.ArgumentParser(
        description="Convert NIFTI file to float64 datatype for cornerstone3D bug reproduction"
    )
    parser.add_argument(
        "input_file", 
        help="Path to input NIFTI file (.nii or .nii.gz)"
    )
    parser.add_argument(
        "output_file", 
        help="Path for output float64 NIFTI file (.nii or .nii.gz)"
    )
    parser.add_argument(
        "--preserve-range", 
        action="store_true",
        help="Preserve original data range (default: convert directly to float64)"
    )
    
    args = parser.parse_args()
    
    try:
        convert_nifti_to_float64(args.input_file, args.output_file)
    except FileNotFoundError:
        print(f"Error: Input file not found: {args.input_file}")
        sys.exit(1)
    except Exception as e:
        print(f"Error converting file: {str(e)}")
        sys.exit(1)


if __name__ == "__main__":
    main()
```

I attach the data for reproduction as gdrive links:

Original volume
https://drive.google.com/file/d/1nDcWU5trJeV3ThG5VCQ9c4sairSnlgFY/view?usp=sharing

Float 64 volume
https://drive.google.com/file/d/1h0wJAIYRr2YsaWqI-2x5N1pGgB8ArOyL/view?usp=sharing


### The expected behavior

It is possible to load and render NIFT2 volumes

### System Information

```bash
System:
    OS: Linux 6.8 Ubuntu 24.04.2 LTS 24.04.2 LTS (Noble Numbat)
    CPU: (16) x64 AMD EPYC 7R32
    Memory: 47.22 GB / 62.10 GB
    Container: Yes
    Shell: 5.2.21 - /bin/bash
  Binaries:
    Node: 22.19.0 - ~/.nvm/versions/node/v22.19.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v22.19.0/bin/yarn
    npm: 10.9.3 - ~/.nvm/versions/node/v22.19.0/bin/npm
    bun: 1.2.21 - /snap/bin/bun
```

--- a/packages/core/src/types/PixelDataTypedArray.ts
+++ b/packages/core/src/types/PixelDataTypedArray.ts
@@ -1,5 +1,6 @@
 export type PixelDataTypedArray =
   | Float32Array
+  | Float64Array
   | Int16Array
   | Uint16Array
   | Uint8Array
@@ -10,6 +11,7 @@ export type PixelDataTypedArray =
 
 export type PixelDataTypedArrayString =
   | 'Float32Array'
+  | 'Float64Array'
   | 'Int16Array'
   | 'Uint16Array'
   | 'Uint8Array'

--- a/packages/core/src/utilities/generateVolumePropsFromImageIds.ts
+++ b/packages/core/src/utilities/generateVolumePropsFromImageIds.ts
@@ -150,6 +150,9 @@ function _determineDataType(
     case 32:
       return 'Float32Array';
 
+    case 64:
+      return 'Float64Array';
+
     default:
       throw new Error(
         `Bits allocated of ${BitsAllocated} is not defined to generate scalarData for the volume.`

--- a/packages/nifti-volume-loader/src/helpers/dataTypeCodeHelper.ts
+++ b/packages/nifti-volume-loader/src/helpers/dataTypeCodeHelper.ts
@@ -17,6 +17,8 @@ export function getArrayConstructor(datatypeCode: number): unknown {
       return Uint16Array;
     case NIFTICONSTANTS.NIFTI_TYPE_UINT32:
       return Uint32Array;
+    case NIFTICONSTANTS.NIFTI_TYPE_FLOAT64:
+      return Float64Array;
     default:
       throw new Error(
         `NIFTI datatypeCode ${datatypeCode} is not yet supported`

--- a/packages/nifti-volume-loader/src/helpers/modalityScaleNifti.ts
+++ b/packages/nifti-volume-loader/src/helpers/modalityScaleNifti.ts
@@ -82,6 +82,10 @@ export default function modalityScaleNifti(
       niiBuffer = new Uint32Array(niftiImageBuffer);
       scalarData = allocateScalarData('Float32Array', niiBuffer);
       break;
+    case NIFTICONSTANTS.NIFTI_TYPE_FLOAT64:
+      niiBuffer = new Float64Array(niftiImageBuffer);
+      scalarData = allocateScalarData('Float64Array', niiBuffer);
+      break;
     default:
       throw new Error(
         `NIFTI datatypeCode ${datatypeCode} is not yet supported`
@@ -150,6 +154,11 @@ function allocateScalarData(
       bitsAllocated = 8;
       checkCacheAvailable(bitsAllocated, nVox);
       scalarData = new Uint8Array(nVox);
+      break;
+    case 'Float64Array':
+      bitsAllocated = 64;
+      checkCacheAvailable(bitsAllocated, nVox);
+      scalarData = new Float64Array(nVox);
       break;
     default:
       throw new Error(`TypedArray ${types} is not yet supported`);

